watertap/tools/oli_api/client.py

```diff
@@ -276,7 +276,11 @@ def dbs_file_cleanup(self, dbs_file_ids=None):
 
         if dbs_file_ids is None:
             dbs_file_ids = self.get_user_dbs_file_ids()
-        r = input(f"WaterTAP will delete {len(dbs_file_ids)} DBS files: [y]/n: ") if self.interactive_mode else ""
+        r = (
+            input(f"WaterTAP will delete {len(dbs_file_ids)} DBS files: [y]/n: ")
+            if self.interactive_mode
+            else ""
+        )
```

Comment:
I'm really not sure, but I think the logic here might not behave the way we want to, i.e. `input()` is run even when `self.interactive_mode` is false-y.

Maybe we can try the following:

```suggestion
        if self.interactive_mode:
            r = input(...)
        else:
            r = ""
```

Reply:
I guess a better solution would be to abstract away the references to `input()` with a dedicated method:

```py
    def _prompt(self, msg, default):
        if self.interactive_mode:
            return input(msg)
        else:
             _logger.info(msg)
             return default
```

And then in the rest of the class, call `self._prompt()` without any further logic needed

Reply:
I added this to `OLIApi` for now at least. Will add it to #1300 for `CredentialManager` as well